biz.aQute.bndlib.tests/src/test/ExportAnnotationTest.java

```diff
@@ -12,11 +14,43 @@
 import aQute.bnd.header.Parameters;
 import aQute.bnd.osgi.Builder;
 import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Packages;
 import junit.framework.TestCase;
 
 @SuppressWarnings("resource")
 public class ExportAnnotationTest extends TestCase {
 
+	public void testExportAnnotationOverlapsExportPackageHeader() throws Exception {
+		try (Builder builder = new Builder()) {
+			builder.addClasspath(new File("bin"));
+
+			builder.setExportPackage("test.export.annotation.testCalculated;version=42");
+			builder.build();
+			Packages exports = builder.getExports();
+			assertThat(exports.size()).isEqualTo(1);
+			Attrs attrs = exports.getByFQN("test.export.annotation.testCalculated");
+
+			// Check override from manifest
+			assertThat(attrs.getVersion()).isEqualTo("42");
+
+			// Check if we still have the stuff on the
+			// annotation
+			assertThat(attrs.get("fizz")).isEqualTo("buzz");
+
+		}
+	}
+
+	public void testRemoveExportAnnotation() throws Exception {
+		try (Builder builder = new Builder()) {
+			builder.addClasspath(new File("bin"));
+
+			builder.setPrivatePackage("test.export.annotation.testCalculated");
+			builder.build();
+			Packages exports = builder.getExports();
+			assertThat(exports.size()).isEqualTo(0);
```

Comment:
This is a problem here. The package is annotated Export and it is not exported. Using Private-Package should not turn off @Export. If you don't want to export a package marked @Export, you should negate it on the Export-Package or -exportcontents instruction.

Reply:
There are valid use cases where you want to ignore the Export annotation. How do you suggest a user can do this?
The -exportcontents is not a good comparison since this is by definition **also** under the control of the bnd file as well. The project owner can do as they see fit but the annotation can come from an external bundle.
So unless there is another mechanism I missed to override the Export annotation than I think this is the right solution.

Reply:
`-exportcontents: !the.package.to.not.export, the.package.to.export` would be how to do it.

Reply:
Also, how would you handle slurping a package from the classpath and then not exporting it?

Reply:
Private-Package is how you populate the bundle. It is common to PP all the contents of a bundle and then list the packages to export on EP. OSGi's build does this. If you want to convert to use `@Export` to export specific packages instead of using EP in the bnd file, then PP cannot cause `@Export` to be ignored.

Reply:
so how do we then do the original use case?

Reply:
Private-Package is how you populate the **private** part. The @Export was designed for people that did not have any Export-Package nor Private-Package. However, once you start to use these headers you must understand what you do.

Reply:
And in your model the use case to slurp but not export is then impossible?

Reply:
no,. This is a real case where the P-P and E-P dual purpose sucks. The P-P really says **private**. Overriding that with an annotation feels wrong because bnd.bnd file always has priority.

Reply:
you should need something like Auto-Package :-(

Reply:
People do

```
Package-Private: ${p}.*
Export-Package: ${p}.api
```
I would like to replace the EP in the bnd file with an `@Export` in the `${p}.api` package. How do I do that?

Reply:
Put nothin in your bnd file?

Reply:
Some of `${p}.*` could come from `-buildpath`.

Reply:
I think it is very bad practice to slurp and automatically export actually. That is very error prone. And export is a long term commitment. I can see 'some' utility in the default model of the project (which was the driver, no E-P, no P-P) but I am pretty strong on that if you start using these headers they should be in control